docs/API-Gateway-CRD-proposal.md

```diff
@@ -0,0 +1,146 @@
+# API Gateway CRD proposal
+
+This document describes proposed API for installing APIGateway component.
+
+## Proposed CR structure
+
+```yaml
+kind: APIGateway
+spec:
+  # Dropped in context of having a single operator
+  #apiRuleControllerConfig:
+  #  defaultCors:
+  #    origins:
+  #      regex:
+  #        - ".*"
+  #    methods:
+  #    - "GET"
+  #    - "POST"
+  #    - "PUT"
+  #    - "DELETE"
+  #    - "PATCH"
+  #    headers:
+  #    - "Authorization"
+  #    - "Content-Type"
+  #    - "*"
+  #  defaultDomain: "kyma-domain.com"
+  #  k8s:
+  #    hpaSpec:
+  #      maxReplicas: 5
+  #      minReplicas: 2
+  #    strategy:
+  #      rollingUpdate:
+  #        maxSurge: 50%
+  #        maxUnavailable: "0"
+  #    resources:
+  #      limits:
+  #        cpu: 250m
+  #        memory: 384Mi
+  #      requests:
+  #        cpu: 10m
+  #        memory: 128Mi
+  
+  # Kyma gateway should always be available
+  #disableKymaGateway: false
+  gateways:
+    - namespace: "some-ns" # Required
+      name: "gateway1" # Required
+      servers:
+        - hosts: # Creating  more than one for the same host:port configuration should result in  Warning
+            - dnsName: "goat.example.com"
```

Comment:
I think "dnsName" can lead to confusion. Also we are using host in the APIRule. Since those two belong together, I think it makes sense to not have them different named. Maybe "hostname" or "host" is a better name? In K8s Gateway API "hostname" is used.

Reply:
The name is from gardener: https://github.com/gardener/external-dns-management/blob/846d76ec5a07ab2b904645bb115bc9ffe5ce239b/pkg/apis/dns/v1alpha1/dnsentry.go#L63C7-L63C7